pkg/kubelet/nodestatus/setters.go

```diff
@@ -197,36 +197,40 @@ func NodeAddress(nodeIP net.IP, // typically Kubelet.nodeIP
 			// For steps 3 and 4, IPv4 addresses are preferred to IPv6 addresses
 			// unless nodeIP is "::", in which case it is reversed.
 			if nodeIPSpecified {
-				ipAddr = nodeIP
+				ipAddr = []net.IP{nodeIP}
 			} else if addr := net.ParseIP(hostname); addr != nil {
-				ipAddr = addr
+				ipAddr = []net.IP{addr}
 			} else {
 				var addrs []net.IP
 				addrs, _ = net.LookupIP(node.Name)
 				for _, addr := range addrs {
 					if err = validateNodeIPFunc(addr); err == nil {
-						if isPreferredIPFamily(addr) {
```

Comment:
if we don't keep the previous behavior this is a breaking change, maybe do this `if !utilfeature.DefaultFeatureGate.Enabled(features.IPv6DualStack)`